### ISSUE TITLE
crash with IllegalArgumentException on every other start

### ISSUE THREAD

**Issue by mi5o** [yanus171/Handy-News-Reader#724]
### Summary

Every time I start HNR, it shows the latest article. When I pick "Mark as unread" from the menu, the application always unconditionally crashes.

### Some background information

Since maybe a year ago, the application changed the start-up behaviour, directly showing the content of the latest article. I did not like it, but I could live with that. The problem was, the article was always somehow broken. Many actions simply did not work, like sharing or reloading the article. As a workaround, I always went to the menu, selected "Mark as unread" and after that, everything worked just fine.

Situation deteriorated some time around late July 2021, or maybe early August. Since then "Mark as unread" causes immediate and unconditional application crash. Luckily, when I close the window detailing the crash circumstances, the application restarts automatically. This time around, it starts with the list of articles. And again, everything works just fine.

### Steps to reproduce

**Pre-conditions:**

- some feeds configured and downloaded
- some articles pre-loaded, but unread
- application expunged from memory by the OS

**Actual actions:**

1. Start the application afresh (it should display the very latest article)
2. Navigate to the article context menu (it should display individual menu items)
3. Select "Mark as unread" option

**Expected result:** The application displays a list of articles, with the latest article marked as unread.

**Actual result:** Application immediately crashes, displaying screen-of-death, with some debugging details.

**Note:** I guess the pre-conditions mentioned above are insufficient. If they were sufficient, probably I won't be the first to complain, more than two months after the issue occurred for the first time. So, most likely, there is something very unique in my set-up, but I have no idea, what it could be.

### Environment

_OS:_ LineageOS 17.1 (based on Android 10)
_Linux:_ 4.14.113
_application:_ HNR 0.16.12 (244), obtained from official F-Droid repo
_SQLite 1:_ 3.22.0 (/system/bin/sqlite3, this is most likely picked up)
_SQLite 2:_ 3.27.2 (/system/xbin/sqlite3)
_special features:_ permissive selinux, rooted, MicroG (services, framework, store and unified NLP) instead G**gle services
_locale_: en-GB

### Configuration Glympse

**databases/FeedEx.db**

- android_metadata: 1 record (locale=en)
- entries: 4091 records
- feeds: 21 records
- filters: 1 record
- label: 1 record
- sqlite_sequence: 5 records (feeds=22, entries=243357, tasks=260035, filters=1, label=1)

**shared_prefs/preferences.xml**

- boolean: 64 records
- float: 1 record
- int: 5 records
- long: 8 records
- string: 35 records

### StackTrace

`java.lang.IllegalArgumentException: Illegal update. Match code=-1; uri=content://ru.yanus171.feedexfork.provider.FeedData/unread_entries/-1
at ru.yanus171.feedexfork.provider.FeedDataContentProvider.update(FeedDataContentProvider.java:657)
at android.content.ContentProvider$Transport.update(ContentProvider.java:420)
at android.content.ContentResolver.update(ContentResolver.java:1993)
at ru.yanus171.feedexfork.fragment.EntryFragment$8.run(EntryFragment.java:699)`

**Comment by yanus171** [comment]
Try https://github.com/yanus171/Handy-News-Reader/issues/683#issuecomment-961809500

**Comment by mi5o** [comment]
I tried.

First I had to uninstall F-Droid build, and then rescue the data into new installation, so things could improve or deteriorate in the meantime, which makes my results a bit less reliable. Sorry about that.

Nevertheless, the application seemed to work without crash, for one and a half day. Not even my own steps to reproduce the issue were helpful. Like if it was fixed for good. Until the application crashed again, when I was not expecting it. I am not 100% sure if this is the same issue, but I seems to me to be very similar if not the same (the line numbers are obviously different).

UNCAUGHT EXCEPTION

```
Stacktrace:
java.lang.IllegalArgumentException: Illegal update. Match code=-1; uri=content://ru.yanus171.feedexfork.provider.FeedData/unread_entries/-1
  at ru.yanus171.feedexfork.provider.FeedDataContentProvider.update(FeedDataContentProvider.java:657)
  at android.content.ContentProvider$Transport.update(ContentProvider.java:420)
  at android.content.ContentResolver.update(ContentResolver.java:1993)
  at ru.yanus171.feedexfork.fragment.EntryFragment$3.run(EntryFragment.java:688)
```

```
Version : 0.16.14
Version Code: 245
Package : ru.yanus171.feedexfork
FilePath : /data/user/0/ru.yanus171.feedexfork/files
Phone Model: SM-G970F
Android Version : 10
User : ivanmeler
Total Internal memory : 118053482496
Available Internal memory : 78098825216
```

Let me stress that no important data were lost due to this issue, I could quickly continue where I wanted, which I truly appreciate. It is mainly an unpleasant user experience.

Other than that, I also observed some regression. My articles used to be preloaded and cached, now they are not, and I must manually force loading each single article. I had no time yet, to dig into the configuration to see whose mistake is this. But the regression occurred without me changing anything in the settings. So I am letting you know this just in case it rings some bell in your mind, without opening a new issue.
